cpp/include/raft/distance/specializations/detail/correlation.cuh

```diff
@@ -18,36 +18,25 @@
 
 #include <raft/distance/detail/distance.cuh>
 
-namespace raft {
-namespace distance {
-namespace detail {
+namespace raft::distance::detail {
+
 extern template void
-distance<raft::distance::DistanceType::CorrelationExpanded, float, float, float, int>(
-  raft::resources const& handle,
-  const float* x,
-  const float* y,
-  float* dist,
-  int m,
-  int n,
-  int k,
-  void* workspace,
-  size_t worksize,
-  bool isRowMajor,
-  float metric_arg);
+  pairwise_matrix_instantiation_point<ops::correlation_distance_op<float, float, int>,
+                                      int,
+                                      float,
+                                      float,
+                                      decltype(raft::identity_op())>(
```

Comment:
Iff you want the type, can't it be used directly? If it can then this will apply in all other places
```suggestion
                                      raft::identity_op>(
```